source/contracts/reporting/ReportingWindow.sol

```diff
@@ -131,10 +135,15 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
         require(markets.contains(_market));
         require(_market.getReportingTokenOrZeroByPayoutDistributionHash(_payoutDistributionHash) == msg.sender);
         IMarket.ReportingState _state = _market.getReportingState();
-        require(_state == IMarket.ReportingState.ALL_REPORTING || _state == IMarket.ReportingState.LIMITED_REPORTING);
+        bool _validReportingState = _state == IMarket.ReportingState.ALL_REPORTING;
+        _validReportingState = _validReportingState || (_state == IMarket.ReportingState.LIMITED_REPORTING);
+        _validReportingState = _validReportingState || (_state == IMarket.ReportingState.DESIGNATED_REPORTING);
```

Comment:
Style: Curious why you split these across multiple statements instead of something like:
```
require(_state == IMarket.ReportingState.ALL_REPORTING
    || _state == IMarket.ReportingState.LIMITED_REPORTING
    || _state == IMarket.ReportingState.DESIGNATED_REPORTING);
```

Reply:
Not a good reason. This is much better!